frontend/src/app/components/shared/song-toolbar/song-toolbar.component.ts

```diff
@@ -0,0 +1,147 @@
+    this.seekSlider = <HTMLInputElement>document.getElementById('seek-slider');
+    this.volumeSlider = <HTMLInputElement>document.getElementById('volume-slider');
+    this.currentTimeContainer = document.getElementById('current-time');
+    this.durationContainer = document.getElementById('duration');
+    this.audio = document.querySelector('audio');
+  }
+
+  UpdateSong = (songInfo: SongInfo) => {
+    this.songForPlay = songInfo;
+    this.resetPlaying();
+    this.audio?.load();
+    this.show = true;
+    this.audio!.loop = this.isRepeating;
+  };
+
+  setTimeChanging = (value: boolean) => {
+    this.isTimeChanging = value;
+  };
+
+  resetPlaying = () => {
+    this.isPlaying = false;
+    this.playPauseButton?.lastElementChild?.classList.replace('pause', 'play');
+  };
+
+  updateTime() {
+    this.displayCurrentTime();
+    this.updateSeekSlider();
+  }
+
+  initSong = () => {
+    this.displayDuration();
+    this.setSeekSliderMax();
+    this.setInitialVolume();
+    this.playPause();
+  };
+
+  setVisibility(show: boolean) {
+    this.show = show;
+  }
+
+  displayDuration = () => {
+    this.durationContainer!.textContent = TimeConverter.secondsToMMSS(this.audio!.duration);
+  };
+
+  displayCurrentTime = () => {
+    this.currentTimeContainer!.textContent = TimeConverter.secondsToMMSS(this.audio!.currentTime);
+  };
+
+  setSeekSliderMax = () => {
+    this.seekSlider?.setAttribute('max', this.audio!.duration.toString());
+  };
+
+  updateSeekSlider = () => {
+    if (!this.isTimeChanging) this.seekSlider!.value = Math.floor(this.audio!.currentTime).toString();
+  };
+
+  getSeekSliderValue = (event: Event) => {
+    this.audio!.currentTime = Number.parseInt((<HTMLInputElement>event.target).value);
+  };
+
+  getVolumeSliderValue = (event: Event) => {
+    const volume = Number.parseInt((<HTMLInputElement>event.target).value);
+    this.savedVolume = volume;
+    this.audio!.volume = volume / 100;
+  };
+
+  setInitialVolume = () => {
+    this.audio!.volume = Number.parseInt(this.volumeSlider!.value) / 100;
+  };
+
+  playPause = () => {
+    if (this.isPlaying) {
+      this.audio?.pause();
+      this.isPlaying = false;
+      this.playPauseButton?.lastElementChild?.classList.replace('pause', 'play');
+    }
+    else {
+      this.audio?.play();
+      this.isPlaying = true;
+      this.playPauseButton?.lastElementChild?.classList.replace('play', 'pause');
+    }
+  };
+
+  muteUnmute = () => {
+    if (this.isMuted) {
+      this.isMuted = false;
+      this.volumeSlider!.value = this.savedVolume.toString();
+      this.audio!.volume = this.savedVolume / 100;
+      console.log(this.audio!.volume);
+    }
+    else {
+      this.isMuted = true;
+      this.volumeSlider!.value = '0';
+      this.audio!.volume = 0;
+    }
+  };
+
+  toggleShuffle = () => {
+    this.isShuffling = !this.isShuffling;
+    (<HTMLButtonElement>document.getElementById('shufflebutton')).classList.toggle('active');
```

Comment:
Use angular for toggling classes

Reply:
Like here? 
https://stackoverflow.com/questions/44535515/angular-ngclass-and-click-event-for-toggling-class

Reply:
Yep